File: APEC.RRHH.api/Core/Models/Employee.cs
using System;
using System.Collections.Generic;
using Core.Contracts;
using Core.Enums;

namespace Core.Models
{
    public sealed class Employee : IEntityBase
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public FeatureStatus Status { get; set; }
        public DateTime AdmissionDate { get; set; }
        public Guid JobId { get; set; }
        public Job Job { get; set; }
        public int MonthlySalary { get; set; }
        public ISet<CandidateInterview> CandidateInterviews { get; set; }
    }
}

File: APEC.RRHH.webapp/src/app/models/employee.ts
import { FeatureStatus } from '../enums/feature-status';
import { UserProfile } from './user-profile';
import { Job } from './job';

export class Employee {
    id: string;
    userId: string;
    user: UserProfile;
    monthlySalary: number;
    jobId: string;
    job: Job;
    admissionDate: Date;
    status: FeatureStatus;
}
